worker/peergrouper/initiate.go

```diff
@@ -50,16 +45,25 @@ func InitiateMongoServer(p InitiateMongoParams) error {
 
 	// Initiate may fail while mongo is initialising, so we retry until
 	// we successfully populate the replicaset config.
-	var err error
-	for attempt := initiateAttemptStrategy.Start(); attempt.Next(); {
-		err = attemptInitiateMongoServer(p.DialInfo, p.MemberHostPort)
-		if err == nil {
-			logger.Infof("replica set initiated")
-			return err
-		}
-		if attempt.HasNext() {
-			logger.Debugf("replica set initiation failed, will retry: %v", err)
-		}
+	retryCallArgs := retry.CallArgs{
+		Clock:       clock.WallClock,
+		MaxDuration: 60 * time.Second,
+		Delay:       5 * time.Second,
+		Func: func() error {
+			return attemptInitiateMongoServer(p.DialInfo, p.MemberHostPort)
+		},
+		NotifyFunc: func(lastError error, attempt int) {
+			logger.Debugf("replica set initiation failed, will retry: %v", lastError)
```

Comment:
This isn't quite the same behavior as before, since it will print "will retry" even when it won't. Which is why I've added line 62

I feel a lie like this isn't too bad, but I would be open to replacing the log message with "might retry" or something along those lines. Thoughts?